src/components/application_manager/rpc_plugins/vehicle_info_plugin/test/vehicle_info_pending_resumption_test.cc

```diff
@@ -0,0 +1,609 @@
+
+  object[strings::params][strings::correlation_id] = corr_id;
+  object[strings::msg_params] =
+      smart_objects::SmartObject(smart_objects::SmartType_Map);
+  return request;
+}
+
+mobile_apis::VehicleDataType::eType ToVDType(const std::string& vd) {
+  if ("gps" == vd) {
+    return mobile_apis::VehicleDataType::VEHICLEDATA_GPS;
+  }
+  if ("speed" == vd) {
+    return mobile_apis::VehicleDataType::VEHICLEDATA_SPEED;
+  }
+  return mobile_apis::VehicleDataType::VEHICLEDATA_OEM_CUSTOM_DATA;
+}
+
+smart_objects::SmartObject CreateVDError(const uint32_t correlation_id,
+                                         const int32_t result_code,
+                                         const std::string& info) {
+  namespace strings = application_manager::strings;
+  namespace hmi_response = application_manager::hmi_response;
+  smart_objects::SmartObject message(smart_objects::SmartType_Map);
+
+  message[strings::params][strings::function_id] =
+      VehicleInfo_SubscribeVehicleData;
+  const auto hmi_protocol_type = 1;
+  message[strings::params][strings::protocol_type] = hmi_protocol_type;
+  message[strings::params][strings::correlation_id] = correlation_id;
+
+  const auto message_type_response = 1;
+  message[strings::params][strings::message_type] = message_type_response;
+  message[strings::params][hmi_response::code] = result_code;
+  message[strings::params][strings::error_msg] = info;
+
+  return message;
+}
+
+typedef std::map<std::string, hmi_apis::Common_VehicleDataResultCode::eType>
+    VDResponseMap;
+smart_objects::SmartObject CreateVDResponse(
+    const hmi_apis::Common_Result::eType common_result,
+    const VDResponseMap& subscriptions_result,
+    uint32_t correleation_id) {
+  namespace strings = application_manager::strings;
+  namespace hmi_response = application_manager::hmi_response;
+  smart_objects::SmartObject message(smart_objects::SmartType_Map);
+  smart_objects::SmartObject params(smart_objects::SmartType_Map);
+  params[strings::function_id] = VehicleInfo_SubscribeVehicleData;
+  const auto message_type_response = 1;
+  params[strings::message_type] = message_type_response;
+  params[strings::correlation_id] = correleation_id;
+  const auto hmi_protocol_type = 1;
+  params[strings::protocol_type] = hmi_protocol_type;
+  params[hmi_response::code] = common_result;
+  message[strings::params] = params;
+  smart_objects::SmartObject msg_params(smart_objects::SmartType_Map);
+  for (const auto& subscription : subscriptions_result) {
+    smart_objects::SmartObject subscription_result(
+        smart_objects::SmartType_Map);
+    subscription_result[strings::data_type] = ToVDType(subscription.first);
+    subscription_result[strings::result_code] = subscription.second;
+    msg_params[subscription.first] = subscription_result;
+  }
+  message[strings::msg_params] = msg_params;
+  return message;
+}
+
+enum class ContainsPolicy { Strict, Nice };
+
+bool CheckThatMessageContainsVD(
```

Comment:
@LuxoftAKutsan maybe it makes sense to move this function to anonymous namespace in the beginning of this file?

Reply:
I have several helper functions for this tests, all of them are defined after test class